spotbugs-exclude.xml

```diff
@@ -0,0 +1,34 @@
+<FindBugsFilter>
+
+	<Match>
+		<Bug pattern="RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE" />
+	</Match>
+
+	<Match>
+		<Bug pattern="URF_UNREAD_PUBLIC_OR_PROTECTED_FIELD" />
+	</Match>
+
+	<Match>
+		<Bug pattern="RV_RETURN_VALUE_IGNORED_BAD_PRACTICE" />
+	</Match>
+
+	<Match>
+		<Bug pattern="URF_UNREAD_FIELD" />
+		<Or>
+			<And>
+				<Class name="io.undertow.server.protocol.ajp.AjpRequestParseState" />
+				<Field name="dataSize" />
+			</And>
+		</Or>
+	</Match>
+
+	<!-- For source/class naming, see https://spotbugs.readthedocs.io/en/stable/filter.html#java-element-name-matching -->
+	<Match>
+		<!-- Ignore all source files with samples in path name -->
```

Comment:
One could argue that the generated code should also be free from issues found by static code analysis. I'm not sure about excluding samples here.

I can see how users have different code standards and running Checkstyle and PMD is not worth the effort, but I'm not sure it is good to exclude Spotbugs here.

Reply:
Another reason to exclude these here is that I assume most users use custom templates. Analyzing our "samples" is therefore very much like analyzing our tests.

Reply:
Another reason to exclude these here is that I assume most users use custom templates. Analyzing our "samples" is therefore very much like analyzing our tests.

Reply:
Another concern is that only a very small subset of our samples are Java :\
Maybe we could later work out the pattern to exclude everything but java client and server outputs in the samples directory.